infrastructure/src/main/java/org/corfudb/infrastructure/management/ClusterRecommendationPolicy.java

```diff
@@ -0,0 +1,35 @@
+package org.corfudb.infrastructure.management;
+
+/**
+ * {@link ClusterRecommendationPolicy} is the enumeration of policies each of which representing
+ * specific algorithms for evaluating the cluster and providing recommendation on the failure or
+ * healing status of servers in a Corfu cluster of servers.
+ *
+ * Created by Sam Behnam on 10/19/18.
+ */
+public enum ClusterRecommendationPolicy {
+    /**
+     * FULLY_CONNECTED_CLUSTER represents a cluster evaluation approach for recommending the
+     * failure and healing status of node in which the implementing algorithm:
+     * Suspect a node of Failed status when the given node is NOT FULLY connected to all the
+     * healthy servers in the cluster.
+     * Suspect a node of Healed status when the given node is FULLY connected to all the healthy
+     * servers in the cluster.
+     *
+     * After applying this policy, the resulting graph of cluster nodes will resemble a Complete
+     * Graph.
+     */
+    FULLY_CONNECTED_CLUSTER,
```

Comment:
I think "strategy" is more descriptive than "policy". 
Also, if we are only implementing one type for now, then what's the point of enumerating other fault detectors that we may implement in the future? Maybe we can get ride of this enum because we only have one failure detector.

Reply:
Strategy makes sense. will change that.

Reply:
Already changed it in the interface patch and once rebase, it will be changed to strategy.